benchmark/aws/instance.py

```diff
@@ -129,13 +129,18 @@ def _get_vpc_info(self, client):
                 vpc_ids += [x['VpcId']]
 
         subnet_ids = {}
-        for x in client.describe_subnets()['Subnets']:
+        for x in client.describe_subnets(
+            Filters=[{
+                'Name': 'cidr-block', 
+                'Values': [ self.settings.subnet_cidr ]
+            }]
+            )['Subnets']:
             if x['VpcId'] in vpc_ids:
                 subnet_ids[x['VpcId']] = x['SubnetId']
 
         sec_group_ids = {}
         for x in client.describe_security_groups()['SecurityGroups']:
-            if x['VpcId'] in vpc_ids:
+            if x['VpcId'] in vpc_ids and x['GroupName'] == f'{self.SECURITY_GROUP_NAME}-{x["VpcId"]}' :
```

Comment:
Here,  in case a VPC has more than one security group, the for loop was picking just the last SG returned by `describe_security_groups()`.  
Indeed, some hosts were being associated to the default VPC Security Group (that AFAIK always exists for every VPC) instead of the security group created. As far as I remember, this was happening for the -0 host, but not for the others. 
Adding a match on the security group name fixed it